src/python/WMCore/Credential/Proxy.py

```diff
@@ -586,10 +586,17 @@ def logonRenewMyProxy(self, proxyFilename=None, credServerName=None):
         # Note that this is saved in a temporary file with the pid appended to the filename. This way we will avoid adding many
         # signatures later on with vomsExtensionRenewal in case of multiple processing running at the same time
         tmpProxyFilename = proxyFilename + '.' + str(os.getpid())
+
+        if self.userName:
```

Comment:
given that I see the same logic in different places in this module, how about we create a method to return the `credname` value? Just call that method every where you need to resolve the user credentials in the same format as: `sha1(self.userDN + "_" + self.myproxyAccount).hexdigest()`

Reply:
@amaltaro I started with the idea of making minimal modifications.

I do not like to introduce a `getCredname` method for something which is not exactly
a credential or anyhow adding a method I can't properly document.
Fact is that clearly different people added code to `Proxy.py` in uncoordinated way,
at times w/o fully understanding what they were doing. Hopefully one day we
will deprecate X509 and this stuff will all go. Hoping that day to be not too far,
I tried to avoid a major overhaul.

But  now that I learnt the inside of this and you made me think more about it, what about 
the following which is a truly minimal change to WMCore Proxy ?

we use `self.userName`  from the argument dictionary passed to
`Proxy.py` as "username" whenever a `myproxy-*` command is used which requires
the `-l username` option (only CRAB use that).
And leave it to the calling CRAB code to decide if to pass a human string or a hash of the DN+ the mysterious `self.myproxyAccount` which when you read this code you have no
clue what it should be ?

AFAIK only CRAB uses `-l username` option in myproxy-* (since it is only used when an indentity switch is needed) and nobody else was using the self.userName, at least not in WMCore repository.  Will also avoid using the name `credname` since this is not really the name of a credential, actually myproxy-* has a credential name argument, via the option `-k`.
I am not sure why the original developers used `-l username` instead of `-k credname` 

This requires to modify the changes I have done to CRAB already and a new round of test from my side, so do not hold your breath... but if you like it more, I'll do.

Reply:
sorry for very logg comment, let me summarize:
if I do what you suggested it will still be a non-understandable hack, which nobody but a CRAB expert can maintain/use, while saving only ~10 lines of code.
Since Proxy.py  will not be there in Run4 for sure, and hopefully gone sooner, I don't see the gain, but I can do it easily.

The alternative is more work for me, but a way to push some (but still not all) CRAB internals knowledge from Proxy.py back into CRAB land.

All in all, having slept on it, I'd rather do what you suggested as it gives me a place where to put a bunch of comments to help others to understand what's going on.

Reply:
Stefano, your idea looks much clean indeed. However, it will also depend on how many clients/places you will have to apply such change. If not too much and you're willing to do so, please go ahead.

BTW, I don't think any of the WMCore services use this module. Even though we could, in order to renew my proxy that gets shipped with the jobs. Anyways, that's not important at the moment and I do not foresee investing time on that.

Reply:
looks like comments crossed. I'go for the simplest.
You will never need to use myproxy to refresh proxy for jobs, condor does it. As long as you have a single user, you only need to keep one valid proxy in each schedd, which I think you do already with the other methods in this class.

Reply:
I decided to follow initial suggestion, but use a method name which makes it clear it is a hack for CRAB and did my best to document it.
Hope you like it @amaltaro 

Do not merge yet. I need to test it now.